apis/src/main/java/xyz/bookself/controllers/user/UserController.java

```diff
@@ -29,9 +26,22 @@ public UserController(UserRepository repository) {
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<User> getUser(@PathVariable Integer id) {
-        final User user = userRepository.findById(id).orElseThrow();
-        return new ResponseEntity<>(user, HttpStatus.OK);
+    public ResponseEntity<User> getUser(@PathVariable Integer id,
+                                        @AuthenticationPrincipal BookselfUserDetails userDetails) {
+        // If nobody is logged in, UNAUTHORIZED
+        if (userDetails == null) {
+            return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+        }
+
+        // If somebody is logged in but is trying to access somebody else's profile, FORBIDDEN
+        if (!userDetails.getId().equals(id)) {
+            return new ResponseEntity<>(HttpStatus.FORBIDDEN);
+        }
+
+        // Otherwise look up the user
+        return userRepository.findById(id)
+                .map(u -> new ResponseEntity<>(u, HttpStatus.OK))
+                .orElseGet(() -> new ResponseEntity<>(HttpStatus.NOT_FOUND));
     }
```

Comment:
@davideengel / @jaredkschreiber can you confirm this approach? The set up is to only allow a user to pull their own details when authenticated.